src/extension/VSProject.cs

```diff
@@ -241,16 +241,41 @@ private void Load()
         /// </summary>
         /// <returns>True if the project was successfully loaded, false otherwise.</returns>
         private bool TryLoadDotNetCoreProject()
-        {
+        {
             XmlNode root = _doc.SelectSingleNode("Project");
 
             if (root != null && SafeAttributeValue(root, "Sdk") != null)
             {
                 string targetFramework = _doc.SelectSingleNode("Project/PropertyGroup/TargetFramework").InnerText;
 
                 XmlNode assemblyNameNode = _doc.SelectSingleNode("Project/PropertyGroup/AssemblyName");
-                // Even console apps are dll's even if <OutputType> has value 'EXE'
-                string assemblyName = assemblyNameNode == null ? $"{Name}.dll" : $"{assemblyNameNode.InnerText}.dll";
+                
+                // Even console apps are dll's even if <OutputType> has value 'EXE',
+                // if TargetFramework is netcore
+                string outputType = "dll";
+
+                if (!targetFramework.StartsWith("netcore"))
+                {
+                    // When targetting standard .Net framework, the default is still dll,
+                    // however, OutputType is now honoured.
+                    // Also if Sdk = 'Microsoft.NET.Sdk.Web' then OutputType default is exe
+                    string sdk = root.Attributes["Sdk"].Value;
+
+                    if (sdk == "Microsoft.NET.Sdk.Web")
+                    {
+                        outputType = "exe";
+                    }
+                    else
+                    {
+                        XmlNode outputTypeNode = _doc.SelectSingleNode("Project/PropertyGroup/OutputType");
+                        if (outputTypeNode != null)
+                        {
+                            outputType = outputTypeNode.InnerText;

```

Comment:
This is correct for exe's, but aren't dll's shown as `<OutputType>Library</OutputType>`?

Would we be safer to do `if (outputTypeNode == "Exe") outputType = "exe" else outputType = "dll"`?

(Or at least the above idea, written in a nicer fashion that I've managed on the train!)

Reply:
I'll look into that.
It might well be an extra case that if you change the output type to dll that it adds this in.
Most of the projects that I made to test against were just all of the different flavours that I could find to create.
I didn't get as far as looking what you might be able to produce in your project file if you messed about with it a lot

Reply:
The three output types I'm familaiar with are `Library`, `Exe` and `WinExe` (to suppress showing the console).